### DISCUSSION TITLE
build python error

### DISCUSSION THREAD

**Discussion by C8504** [google/or-tools#3353] [Build and configuration issues] [answered]
Hi Experts,

When I build python on main branch, I got below error message, could you please help me to resolve? thanks.

       "C:\or-tools\build_make\install.vcxproj" (default target) (1) ->
       "C:\or-tools\build_make\ALL_BUILD.vcxproj" (default target) (3) ->
       "C:\or-tools\build_make\python_package.vcxproj" (default target) (101) ->
       "C:\or-tools\build_make\Pyortools_proto.vcxproj" (default target) (104) ->
       (CustomBuild target) ->
         C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: Custom build for 'C:\or-tools\build_make
       \CMakeFiles\5ccd2c9c83dec2f4a839c84a257672eb\bop_parameters_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\6cb39017a78a2907bc7dafc5b958c9ca\assignment_pb2.py.rule;C:\or-tools\build_m
       ake\CMakeFiles\6cb39017a78a2907bc7dafc5b958c9ca\routing_enums_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\6cb39017a78a2907bc7dafc5b958c9ca\routing_parameters_pb2.py.rule;C:\or-too
       ls\build_make\CMakeFiles\6cb39017a78a2907bc7dafc5b958c9ca\search_limit_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\6cb39017a78a2907bc7dafc5b958c9ca\search_stats_pb2.py.rule;C:\or-
       tools\build_make\CMakeFiles\6cb39017a78a2907bc7dafc5b958c9ca\solver_parameters_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\1014b45ee9333c6faec7ff107197d608\parameters_pb2.py.rule;
       C:\or-tools\build_make\CMakeFiles\d655198a4283f90eb21c297d7ebb8167\flow_problem_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\41a2462f6597ad9fc0080add51ae0db9\linear_solver_pb2.py.r
       ule;C:\or-tools\build_make\CMakeFiles\cca4450b59f06bb8ca2a680233bab48f\multiple_dimensions_bin_packing_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\cca4450b59f06bb8ca2a680233bab48f
       \vector_bin_packing_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\a176e8579ec23b56167be2a64a9f44a8\solve_log_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\a176e8579ec23b56167be2a64a
       9f44a8\solvers_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\ce0d0b16e1b3e763ef8579c1055818a1\boolean_problem_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\ce0d0b16e1b3e763ef8579c10
       55818a1\cp_model_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\ce0d0b16e1b3e763ef8579c1055818a1\cp_model_service_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\ce0d0b16e1b3e763ef8579
       c1055818a1\sat_parameters_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\b89ff7b3ac8e9ce643ccda7eda29c80a\course_scheduling_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\b89ff7b3ac8e
       9ce643ccda7eda29c80a\jobshop_scheduling_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\b89ff7b3ac8e9ce643ccda7eda29c80a\rcpsp_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\09f29eabc9
       a4fa6c362bcb527cd2202f\optional_boolean_pb2.py.rule;C:\or-tools\build_make\CMakeFiles\dcfceab16efa691335b8abefd9073677\Pyortools_proto.rule' exited with code 1. [C:\or-tools\build_m
       ake\Pyortools_proto.vcxproj]

    164 Warning(s)
    1 Error(s)

**Comment by Mizux** [comment, marked as answer]
Do you have other trace ? I guess the error was above.

Which build command did you use ?

**Reply by C8504** [reply]
I used below command.
tools\make.exe python

**Reply by C8504** [reply]
After I added \Python\Python38\Scripts to PATH, it works well now.
